src/python/correct_fastq.py

```diff
@@ -85,15 +86,41 @@ def process_fastqs(input_r1_fastq_file, input_r2_fastq_file,
     """
     # QC counters
     cellbarcode_match = cellbarcode_mismatch = cellbarcode_poly_g = read2_start_poly_g = 0
+
+    read1_out_writer = xopen.xopen(output_read1_fastq_file, mode = 'w')
+    read2_out_writer = xopen.xopen(output_read2_fastq_file, mode ='w')
+
+    buffer1 = deque()
+    buffer2 = deque()
+    buffer_counter = 0
     
     # process FASTQs together
-    with dnaio.open(input_r1_fastq_file, input_r2_fastq_file) as reader, \
-         dnaio.open(output_r1_fastq_file, output_r2_fastq_file, mode="w") as writer:
-             
-        for read_1, read_2 in reader:
+    with xopen.xopen(input_read1_fastq_file, mode= "r", threads= 8) as read1_fh, xopen.xopen(input_read2_fastq_file, mode= "r", threads= 8) as read2_fh:
```

Comment:
threads should probably be a input variable

Reply:
but we can have it as an issue for the future